exonum/src/storage/proof_list_index/mod.rs

```diff
@@ -377,21 +373,18 @@ where
     /// let list_proof = index.get_range_proof(1, 3);
     /// ```
     pub fn get_range_proof(&self, from: u64, to: u64) -> ListProof<V> {
-        if to > self.len() {
-            panic!(
-                "Illegal range boundaries: the len is {:?}, but the range end is {:?}",
-                self.len(),
-                to
-            )
-        }
         if to <= from {
             panic!(
                 "Illegal range boundaries: the range start is {:?}, but the range end is {:?}",
                 from, to
             )
         }
 
-        self.construct_proof(self.root_key(), from, to)
+        if to > self.len() {
+            ListProof::Absent(to, self.list_hash())
```

Comment:
This feature should be documented as well. It's not self-evident that we should return a proof of absence of the last element in range if it does not overlap with valid indices.